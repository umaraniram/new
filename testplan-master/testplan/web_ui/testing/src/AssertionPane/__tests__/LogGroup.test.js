import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import LogGroup from '../LogGroup';

function defaultProps() {
  return {
    logs: [
        {
            "uid":"04b2be21-f14c-4c39-917f-e3e9f7b3eec3",
            "created":"2019-11-28T03:14:37.447875+00:00",
            "funcName":"__exit__",
            "levelno":40,
            "lineno":123,
            "message":"Traceback (most recent call last):\n  File \"testplan.py\", line 123, in _run_testcase(\n        <testplan.testing.multitest.base.MultiTest object at 0x7fb279443320>,\n        <bound method BasicSuite.abc of <__main__.BasicSuite object at 0x7fb279443080>>,\n        None,\n        None,\n        TestCaseReport(name=\"abc\", id=\"abc\", entries=[]))\n    testcase(self.resources, case_result)\n  File \"test_plan.py\", line 16, in abc(<__main__.BasicSuite object at 0x7fb279443080>, Environment[[]], [])\n    print(1/0)\nZeroDivisionError: division by zero",
            "levelname":"ERROR"
        }
    ],
    index: 0,
    onClick: jest.fn()
  };
}


describe('LogGroup', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<LogGroup {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});
