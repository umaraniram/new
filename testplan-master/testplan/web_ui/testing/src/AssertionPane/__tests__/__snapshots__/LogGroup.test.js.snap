// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogGroup shallow renders the correct HTML structure 1`] = `
<div
  className="LogGroup"
>
  <div
    key="04b2be21-f14c-4c39-917f-e3e9f7b3eec3"
  >
    <pre
      className="logEntry_137u7ef"
    >
      Traceback (most recent call last):
  File "testplan.py", line 123, in _run_testcase(
        &lt;testplan.testing.multitest.base.MultiTest object at 0x7fb279443320&gt;,
        &lt;bound method BasicSuite.abc of &lt;__main__.BasicSuite object at 0x7fb279443080&gt;&gt;,
        None,
        None,
        TestCaseReport(name="abc", id="abc", entries=[]))
    testcase(self.resources, case_result)
  File "test_plan.py", line 16, in abc(&lt;__main__.BasicSuite object at 0x7fb279443080&gt;, Environment[[]], [])
    print(1/0)
ZeroDivisionError: division by zero
    </pre>
  </div>
</div>
`;
